src/Hooks/Dispatchers/ReplaceCommandHookDispatcher.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Consolidation\AnnotatedCommand\Hooks\Dispatchers;
+
+use Consolidation\AnnotatedCommand\CommandData;
+use Consolidation\AnnotatedCommand\Hooks\HookManager;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Event\ConsoleCommandEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Call hooks.
+ */
+class ReplaceCommandHookDispatcher extends HookDispatcher
+{
+
+    /**
+     * @return int
+     */
+    public function hasReplaceCommandHook()
+    {
+        return count($this->getReplaceCommandHooks());
+    }
+
+    /**
+     * @return \callable[]
+     */
+    public function getReplaceCommandHooks()
+    {
+        $hooks = [
+            HookManager::REPLACE_COMMAND_HOOK,
+        ];
+        $replaceCommandHooks = $this->getHooks($hooks);
+
+        return $replaceCommandHooks;
+    }
+
+    /**
+     * @param \Consolidation\AnnotatedCommand\CommandData $commandData
+     *
+     * @return callable
+     */
+    public function getReplacementCommand(CommandData $commandData)
+    {
+        $replaceCommandHooks = $this->getReplaceCommandHooks($commandData);
+
+        // We only take the first hook implementation of "replace-command" as the replacement. Commands shouldn't have
+        // more than one replacement.
```

Comment:
I wonder what the right UX should be if there are multiple replacements for the same command. Example: using an application that supports plugins, user installs two independently-developed plugins with overlapping functionality, both of which replace the same command. Perhaps we should simply log a warning when this happens, and print out the class::method names of every replacement for the given function. This would be as good a place as any to do that -- the desired behavior being that you don't get the warning unless you try to run a command with multiple replacement options.